ADRs/RH/SIG-SRE/ADR-00002 Open Source Documentation.md

```diff
@@ -0,0 +1,59 @@
+# ADR-00002 Open Source Documentation
+
+Authors: Rachel Lawton, David Martin
+
+## Status
+
+Draft.
+
+## Problem Statement
+
+We want to build an upstream community around managed services. This means all aspects of the managed service including the ability to run and manage the service, which extends to operational knowledge & procedures. One such element is Standard Operating Procedures (SOPs). These SOPs include information like how to scale up the service as load increases as well as how to diagnose and fix specific issues based on alerting. However, currently these SOPs have been written from an internal perspective and are not immediately usable or consumable upstream.
+This document proposes a solution for maintaining an upstream version of the SOPs and having a mechanism for consuming them downstream, as well as vice versa.
+
+## Goals
+
+* Propose a model for maintaining 2 versions of documents, for upstream and downstream consumption
+* Provide a worked example of this model as a reference
+
+## Non-goals
+
+* Implement a complete solution for a product or service
+
+## Current Architectures
+
+A Standard Operating Procedure (SOP) is a document used by SREs in response to alerts and other events.
+An example of what a current SOP architecture looks like is the Red Hat OpenShift Streams for Apache Kafka (RHOSAK) managed service. The RHOSAK SOPs are located in a private repository, as the SOPs contain references to Red Hat specific internal information that is not suitable for consumption by external users. Such information includes links to Vault secrets, information about how RHOSAK specifically set up their stage and production environments and information on the infrastructure itself, for example what certain custom resources should look like. The flow of how information is contributed to the RHOSAK SOPs is for an engineer or SRE to submit PRs to the repo.
+
+## Proposed Architecture
+
+The proposed solution is to follow a two repository model. One private repository that will contain SOPs that have the Red Hat internal information mentioned above, the other being a public repository that will have SOPs with the Red Hat internal information removed. This model will allow the SRE teams to use the private SOP repository without any information loss and allow upstream users to use and contribute to the SOP without breaching any Red Hat data policies.
```

Comment:
Thank you for proposing this architecture! Have we already considered an overall logical split of documentation instead of filtered mirroring? 

For example, having an internal/private 'runbook' for incidents, with links to external SOP docs. In the case of many upstream projects, there's an ongoing effort to publish runbooks for the alerts that ship with the project. Here's a good one: https://runbooks.prometheus-operator.dev

So in this case, we could look at it this way:
* The 'internal' runbook would have information on specifics for our implementation of the upstream project, access details, configuration documentation etc. 
* The 'upstream' SOP's will be used exclusively for troubleshooting and resolution steps. 

This way, we would adhere to an 'upstream-first' principle and contribute directly to the upstream repositories instead of having to do a manual filter/processing of the runbooks. 

What do you think?